compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -477,8 +478,8 @@ pub fn normalize_param_env_or_error<'tcx>(
 /// hold. Used when creating vtables to check for unsatisfiable methods.
 pub fn impossible_predicates<'tcx>(tcx: TyCtxt<'tcx>, predicates: Vec<ty::Clause<'tcx>>) -> bool {
     debug!("impossible_predicates(predicates={:?})", predicates);
-
-    let infcx = tcx.infer_ctxt().build();
+    // FIXME(@lcnr): Why do we check these in post-analysis mode?
```

Comment:
I'm confused why this isn't obvious -- this is only used for codegen.

Reply:
Perhaps this function should be renamed `impossible_predicates_for_codegen` or something.

Reply:
Oh, and also used in `mir_drops_elaborated_and_const_checked` to taint a body that has impossible preds -- but that's also post-analysis. And I wouldn't be sad if we made that a separate check...